railgun/manager/manager.go

```diff
@@ -237,14 +240,26 @@ func Run(ctx context.Context, c *Config) error {
 		setupLog.Error(err, "cannot create a Kong Admin API client")
 	}
 
-	kongClient, err := kong.NewClient(&c.KongURL, httpclient)
+	kongClient, err := kong.NewClient(&c.KongAdminURL, httpclient)
 	if err != nil {
 		setupLog.Error(err, "unable to create kongClient")
 		return err
 	}
+	if c.KongWorkspace != "" {
```

Comment:
To minimize the amount of logic in `main` and only perform wiring here, please wrap that in a function:

```go
func getKongClientForWorkspace(adminURL string, workspace string, httpclient *http.Client) (*kong.Client, error) {
  if workspace == "" {
    return kong.NewClient(&c.KongAdminURL, httpclient)
  }
  if err := /* check if workspace exists */; err != nil {
    return nil, fmt.Errorf("failed to ensure workspace in kong: %w", err)
  }
  return kong.NewClient(kong.String(c.KongAdminURL+"/"+c.KongWorkspace), httpclient)
}
```

and use it in a minimalistic fashion like this:

```go
kongClient, err := getKongClientForWorkspace(&c.KongAdminURL, &c.KongWorkspace, httpclient)
if err != nil {
  setupLog.Error(err, "unable to create kongClient")
  return err
}
```

Reply:
2a5b053